src/components/RecordTroubleshootDataToolMenu/handleStopRecording.android.ts

```diff
@@ -0,0 +1,32 @@
+import RNFetchBlob from 'react-native-blob-util';
+import type StopRecordingParams from './handleStopRecording.types';
+
+export default async function handleStopRecording({
+    profilingData,
+    infoFileName,
+    appInfo,
+    logsWithParsedMessages,
+    onDisableLogging,
+    cleanupAfterDisable,
+    zipRef,
+    onDownloadZip,
+    setProfileTracePath,
+}: StopRecordingParams): Promise<void> {
+    const {profilePath} = profilingData;
+
+    if (profilePath) {
+        try {
+            // Check if it is an internal path of `DownloadManager` then append content://media to create a valid url
+            const {path} = await RNFetchBlob.fs.stat(!profilePath.startsWith('content://media/') && profilePath.match(/\/downloads\/\d+$/) ? `content://media/${profilePath}` : profilePath);
+            setProfileTracePath?.(path);
+        } catch {
+            setProfileTracePath?.(profilePath);
+        }
+    }
+
+    zipRef.current?.file(infoFileName, appInfo);
```

Comment:
```js
zipRef.current?.file(infoFileName, appInfo);
await onDisableLogging(logsWithParsedMessages);
cleanupAfterDisable();
onDownloadZip?.();
```

This is repeated 3 times, would it be possible to move it to a separate function? Or something like `onSuccess` callback we pass to `handleStopRecording`